server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -271,6 +261,16 @@ public final class IndexSettings {
         Property.Final,
         Property.IndexScope
     );
+
+    public static final TimeValue DEFAULT_REFRESH_INTERVAL = new TimeValue(1, TimeUnit.SECONDS);
+    public static TimeValue STATELESS_MIN_NON_FAST_REFRESH_INTERVAL = TimeValue.timeValueSeconds(5);
+    public static final Setting<TimeValue> INDEX_REFRESH_INTERVAL_SETTING = Setting.timeSetting("index.refresh_interval", (settings) -> {
+        if (EXISTING_SHARDS_ALLOCATOR_SETTING.get(settings).equals("stateless") && INDEX_FAST_REFRESH_SETTING.get(settings) == false) {
```

Comment:
It seems possible to test this in this PR, can we do that (unless it is too hard)?

Reply:
Indeed, I added unit tests in this PR.